openpype/hosts/maya/plugins/publish/validate_gltf_textures_names.py

```diff
@@ -0,0 +1,136 @@
+        for shader in main_shader:
+            albedo = cmds.listConnections(shader + ".TEX_color_map")[0]
+            dif_path = cmds.getAttr(albedo + ".fileTextureName")
+            dif = dif_path.split(".")[0]
+            # "_D"
+            if not dif.endswith("_D"):
+                invalid.add(dif_path)
+            orm_packed = cmds.listConnections(shader + ".TEX_ao_map")
+            if orm_packed:
+                orm_path = cmds.getAttr(orm_packed[0] + ".fileTextureName")
+                orm = orm_path.split(".")[0]
+                # "_ORM"
+                if not orm.endswith("_ORM"):
+                    invalid.add(orm_path)
+            nrm = cmds.listConnections(shader + ".TEX_normal_map")
+            if nrm:
+                nrm_path = cmds.getAttr(nrm[0] + ".fileTextureName")
+                nrm_map = nrm_path.split(".")[0]
+                # "_N"
+                if not nrm_map.endswith("_N"):
+                    invalid.add(nrm_path)
+
+        return list(invalid)
+
+    def get_texture_node_invalid(self, instance):
+        invalid = set()
+        shading_grp = self.get_material_from_shapes(instance)
+        main_shader = cmds.listConnections(shading_grp,
+                                           destination=True,
+                                           type="StingrayPBS")
+        for shader in main_shader:
+            # diffuse texture file node
+            albedo = cmds.listConnections(shader + ".TEX_color_map")
+            if not albedo:
+                invalid.add(albedo)
+            orm_packed = cmds.listConnections(shader + ".TEX_ao_map")
+            if not orm_packed:
+                continue
+            nrm = cmds.listConnections(shader + ".TEX_normal_map")
+            if not nrm:
+                continue
+        return list(invalid)
+
+    def get_texture_shader_invalid(self, instance):
+
+        invalid = set()
+        shading_grp = self.get_material_from_shapes(instance)
+        material_name = "{}.surfaceShader".format(shading_grp)
+        material = cmds.listConnections(material_name,
+                                        source=True,
+                                        destination=False,
+                                        type="StingrayPBS")
+
+        if not material:
+            # add material name
+            material = cmds.listConnections(material_name)[0]
+            invalid.add(material)
+        return list(invalid)
+
+    def get_material_from_shapes(self, instance):
+        shapes = cmds.ls(instance, type="mesh", long=True)
+        for shape in shapes:
+            shading_grp = cmds.listConnections(shape,
+                                               destination=True,
+                                               type="shadingEngine")
+
+            if shading_grp:
+                return shading_grp[0]
+            else:
+                raise RuntimeError("No Shading Engine found in the scene")
```

Comment:
This logic will only ever return the material for the FIRST SHAPE in `get_material_from_shapes`. I think it's a bit too big of an assumption that there will only ever be one material assigned to multiple meshes.

This would return all shading engines on the selected objects:
```python
    def get_shading_engines_from_meshes(self, instance):
        shapes = cmds.ls(instance, type="mesh", long=True)
        shading_engines = cmds.listConnections(shapes,
                                               destination=True,
                                               type="shadingEngine")
        return shading_engines or []
```
_(Technically it would return any shadingEngine that is connected as an output from the given mesh shape - _it could potentially still NOT be assigned as a shader - it's an assumption that the shading engine is only ever connected to the mesh if were connected as a set member)_

It would not detect if any of the shapes did NOT have a shader assigned. (Not sure if this validator should check that?)